src/cfnlint/__init__.py

```diff
@@ -873,6 +877,131 @@ def check_value(self, obj, key, path,
 
         return matches
 
+    def is_resource_available(self, path, resource):
+        """
+            Compares a path to resource to see if its available
+            Returns scenarios that may result in the resource doesn't exist
+        """
+        results = []
+        path_conditions = self.get_conditions_from_path(self.template, path)
+        resource_condition = self.template.get('Resources', {}).get(resource, {}).get('Condition')
+        if resource_condition:
+            # resource conditions are always true.  If the same resource condition exists in the path
+            # with the same True value then nothing else matters
+            test_path_conditions = copy(path_conditions)
+            if not test_path_conditions.get(resource_condition):
+                test_path_conditions[resource_condition] = set()
+            if not test_path_conditions.get(resource_condition):
```

Comment:
Dupe of L892, and L893 should set this if not already present in L892?

Reply:
agreed.  Will be fixed.